# Python Exceptions Tasks :happy:

This repository contains solutions to the Python Exceptions Tasks. The tasks invlove working with exceptions in Python and implementing various functions to handle exceptions in different scenarios.

## List of Tasks
1. Safe list Printing: Implement a function that prints a specified number of elements from a list.

2. Safe printing of an integer list: Implement a function that prints an integer value.

3. Print and count integers: Implement a function that prints the first x elements of a list, considering only integers.

4. Integers division with debug: Implement a function that divides two integers and prints the resuls.

5. Divide a list: Implement a function that divides two lists elements by elements

6. Raise exception: Implement a function that raises a type exception.

7. Raise a message: Implement a function that raises a name exception with a message

8. Safe integer print with error message: Implement a function that prints an integer and handles errors, displaying an error message.

9. Safe function: Implement a function that executes another function safely, handling any exceptions that may occur

10. ByteCode->Python #4: Implement a Python function based on a given Python bytecode.


## Getting Started

To run the tasks, follow these steps:
1. Clone the repository: `git clone <repository-url>`

2. Navigate to the project directory: `cd 0x05-python-exceptions`

3. Execute the desired task by running the corresponding Python file: `python3 <filename.py>`


## Requirements

* Python 3.x

## File Structure

The repository has the following file structure:

```
0x05-python-exceptions/
├── 0-safe_print_list.py
├── 1-safe_print_integer.py
├── 2-safe_print_list_integers.py
├── 3-safe_print_division.py
├── 4-list_division.py
├── 5-raise_exception.py
├── 6-raise_exception_msg.py
├── 100-safe_print_integer_err.py
├── 101-safe_function.py
├── 102-magic_calculation.py
├── README.md
└── tests/
    ├── test_0-safe_print_list.py
    ├── test_1-safe_print_integer.py
    ├── test_2-safe_print_list_integers.py
    ├── test_3-safe_print_division.py
    ├── test_4-list_division.py
    ├── test_5-raise_exception.py
    ├── test_6-raise_exception_msg.py
    ├── test_100-safe_print_integer_err.py
    ├── test_101-safe_function.py
    ├── test_102-magic_calculation.py
    └── __init__.py

```

The `test/` directory contains test files for each task to verify the correctness of the implementations

## Testing
To run the tests for the tasks, execute the following command from the project directory:

```
python -m unittest discover tests
```

The command will run test files and display the test results.


## Contributing

Contributions to this repository are welcome. If you find any issues or want to suggest improvements, please create an issue or submit a pull request.


## License

This project is licensed under the MIT License. See the `Licence` file for more information.
